### PR TITLE
S3DMFT-2799 Manage Associations with Medical Staff Practitioners

### PR DESCRIPTION
**Pull request by shruthi-rapala**
- Added table to show medical staff association
- Added Component to create medical staff association
- Added component to edit medical staff association
- Added component to manage/edit medical staff association
- Updated Dialog header styles

### CODE REVIEW

File: medical-portal/src/UI/medical-portal/src/app/user-profile/edit-medical-staff-association-dialog/edit-medical-staff-association-dialog.component.html [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+              />
+            </mat-form-field>
+          </div>
+        </div>
+        <div>
+          <label class="form-label">Select Medical Practitioner(s) to associate the MOA with:*</label>
+          <mat-selection-list #practitioners>
+            <mat-list-option
+              checkboxPosition="before"
+              *ngFor="let practiotioner of listOfPractitioners"
```
Comment (GeorgeWalker) [inline medical-portal/src/UI/medical-portal/src/app/user-profile/edit-medical-staff-association-dialog/edit-medical-staff-association-dialog.component.html:55]: typo

File: medical-portal/src/UI/medical-portal/src/app/user-profile/manage-medical-staff-association-dialog/manage-medical-staff-association-dialog.component.html [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  </button>
+</div>
+<div mat-dialog-content>
+  <!-- <h1 mat-dialog-title class="modal-title">Edit Profile</h1> -->
+  <div mat-dialog-content class="dialog-content">
+    <form>
+      <div class="container">
+        <h1>Are you sure you want to remove the following association?</h1>
+        <p>
+          You are about to remove the association between <b>Sharon Torres</b> and the following medical practitioner.
```
Comment (GeorgeWalker) [inline medical-portal/src/UI/medical-portal/src/app/user-profile/manage-medical-staff-association-dialog/manage-medical-staff-association-dialog.component.html:14]: Should be able to have an input parameter for the name here

File: medical-portal/src/UI/medical-portal/src/app/user-profile/manage-medical-practitioner-association-dialog/manage-medical-practitioner-association-dialog.component.ts [hunk 3]
```diff
@@ -2,27 +2,27 @@ import { Component, OnInit } from '@angular/core';
...
-export class ManageMedicalPractitionerAssociationDialogComponent{
+export class ManageMedicalPractitionerAssociationDialogComponent {
   displayedColumns: string[] = ['fullName', 'role', 'lastActive'];
   dataSource = [
-    { id: "1", fullName: "Dr. Rajan Mehra", role: "Medical Practioner", lastActive: "October 5, 2021" },
+    {
+      id: '1',
+      fullName: 'Dr. Rajan Mehra',
+      role: 'Medical Practitioner',
+      lastActive: 'October 5, 2021',
```
Comment (GeorgeWalker) [inline medical-portal/src/UI/medical-portal/src/app/user-profile/manage-medical-practitioner-association-dialog/manage-medical-practitioner-association-dialog.component.ts:18]: Is this meant to be the current user?  If so you should be able to use the users/current service to get this.  It is probably also in the page context

Comment (shruthi-rapala) [reply medical-portal/src/UI/medical-portal/src/app/user-profile/manage-medical-practitioner-association-dialog/manage-medical-practitioner-association-dialog.component.ts:18]: It is not the current user it is the selected Medical Practitioner Association.

**Review by GeorgeWalker — APPROVED** [review]

**Pull request merged**